--- urpmi-7.25/urpm/select.pm.orig	2013-10-22 23:46:00.462053052 +0100
+++ urpmi-7.25/urpm/select.pm	2013-10-22 23:46:50.519055078 +0100
@@ -8,7 +8,8 @@
 use urpm::sys;
 use URPM;
 
-my $default_priority_list = 'rpm,perl-URPM,perl-MDV-Distribconf,urpmi,meta-task,glibc,aria2,systemd';
+my $default_priority_list = 'rpm,perl-RPMBDB,perl-URPM,perl-MDV-Distribconf,urpmi,meta-task,glibc,aria2,systemd';
+my $default_priority_list = 'rpm,rpm-build,perl-RPMBDB,perl-URPM,perl-MDV-Distribconf,urpmi,meta-task,glibc,aria2,systemd,lib64systemd-daemon-devel,lib64systemd-login-devel.libsystemd-daemon-devel,libsystemd-login-devel';
 my @priority_list = split(',', $default_priority_list);
 
 my $evr_re = qr/[^\-]*-[^\-]*\.[^\.\-]*$/;
