diff -Naur urpmi-8.03.2/urpm/select.pm urpmi-8.03.2.tpg/urpm/select.pm
--- urpmi-8.03.2/urpm/select.pm	2015-01-11 12:48:24.000000000 +0000
+++ urpmi-8.03.2.tpg/urpm/select.pm	2015-01-18 00:42:11.187250178 +0000
@@ -8,7 +8,7 @@
 use urpm::sys;
 use URPM;
 
-my $default_priority_list = 'rpm,perl-RPMBDB,perl-URPM,perl-MDV-Distribconf,perl-File-Sync,urpmi,meta-task,glibc,aria2,systemd';
+my $default_priority_list = 'setup,filesystem,rpm,perl-RPMBDB,perl-URPM,perl-MDV-Distribconf,perl-File-Sync,urpmi,meta-task,glibc,aria2,systemd';
 my @priority_list = split(',', $default_priority_list);
 
 my $evr_re = qr/[^\-]*-[^\-]*\.[^\.\-]*$/;
