--- urpmi-7.25/urpm/mirrors.pm.orig	2013-09-18 21:31:15.412227222 +0100
+++ urpmi-7.25/urpm/mirrors.pm	2013-09-18 21:31:26.799227682 +0100
@@ -334,7 +334,7 @@
     # a potential attacker from redirecting to his own "update" server.
     # The server already supports https access, but the client side needs to
     # work out certificate verification.
-    if ($product_id->{branch} == "Devel") {
+    if ($product_id->{branch} eq "Devel") {
         "http://downloads.openmandriva.org/mirrors/cooker.$arch.list"
           . (@para ? '?' . join('&', @para) : '');
     } else {
