diff -Naur urpmi-7.29.1/urpm/select.pm urpmi-7.29.1.tpg/urpm/select.pm
--- urpmi-7.29.1/urpm/select.pm	2014-11-11 14:12:58.000000000 +0000
+++ urpmi-7.29.1.tpg/urpm/select.pm	2014-11-12 00:06:36.638652660 +0000
@@ -371,7 +371,7 @@
 		$need_restart = _resolve_priority_upgrades($urpm, $db, $state, $state->{selected}, \@l, %options);
 	    }
 	}
-	$urpm->{options}{'split-length'} = 0 if $need_restart;
+#	$urpm->{options}{'split-length'} = 0 if $need_restart;
     }
     $need_restart;
 }
