--- urpmi-6.70.orig/urpm/mirrors.pm	2011-07-24 18:54:53.000000000 +0400
+++ urpmi-6.70/urpm/mirrors.pm	2012-08-06 01:39:02.440037463 +0400
@@ -304,7 +304,7 @@
     my @para = grep { $_ } $ENV{URPMI_ADDMEDIA_REASON};
     my $product_version = $ENV{URPMI_ADDMEDIA_PRODUCT_VERSION} || $product_id->{version};
 
-    "https://api.mandriva.com/mirrors/$product_type.$product_version.$arch.list"
+    "http://mirror.rosalinux.com/rosa/rosa$product_version/rosa$product_version.$arch.list"
       . (@para ? '?' . join('&', @para) : '');
 }
