--- urpmi-6.70.orig/urpm/media.pm	2012-01-10 19:34:43.000000000 +0400
+++ urpmi-6.70/urpm/media.pm	2012-08-06 00:15:18.970250325 +0400
@@ -1020,6 +1020,39 @@
                 $add_by_default = 0;
             }
         }
+        
+#For ExtendedEdition need be enable non-free and restricted
+	if ($product_id->{product} eq 'EE') {
+             if (member('non-free', @media_types)) {
+				 $urpm->{log}(N("enabling non-free medium `%s'", $media));
+				 $add_by_default = 1; }
+             if (member('restricted', @media_types)) {                 
+                $urpm->{log}(N("enabling restricted medium `%s'", $media)); 
+            	 $add_by_default = 1; }
+             if (member('contrib', @media_types)) {
+				 $urpm->{log}(N("enabling contrib medium `%s'", $media));
+				 $add_by_default = 1; }
+        }
+
+#For Community Edition need be enable non-free, contrib and restricted
+        if (($product_id->{product} eq 'CE') || ($product_id->{product} eq 'GE') || ($product_id->{product} eq 'LXDE')) {
+            if (member('non-free', @media_types)) {
+                $urpm->{log}(N("enabling non-free medium `%s'", $media));
+                $add_by_default = 1;
+            }
+        }
+        if (($product_id->{product} eq 'CE') || ($product_id->{product} eq 'GE') || ($product_id->{product} eq 'LXDE')) {
+            if (member('contrib', @media_types)) {
+                $urpm->{log}(N("enabling contrib medium `%s'", $media));
+                $add_by_default = 1;
+            }
+        }
+        if (($product_id->{product} eq 'CE') || ($product_id->{product} eq 'GE') || ($product_id->{product} eq 'LXDE')) {
+            if (member('restricted', @media_types)) {
+                $urpm->{log}(N("enabling restricted medium `%s'", $media));
+                $add_by_default = 1;
+             }
+         }        
 	my $ignore;
         if ($options{ask_media}) {
             $options{ask_media}->($media_name, $add_by_default) or next;
